tests/main/auto-refresh-pre-download/task.yaml

```diff
@@ -0,0 +1,109 @@
+summary: Check that an inhibited auto-refresh triggers a pre-download change and resumes on close
+
+# Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+systems: [-ubuntu-14.04-*]
+
+kill-timeout: 5m
+
+environment:
+  REFRESH_TYPE/close: "close"
+  REFRESH_TYPE/ignore: "ignore-running"
+
+prepare: |
+  snap install --devmode jq
+  snap install test-snapd-sh
+  snap refresh --unhold
+
+restore: |
+  snap remove --purge jq || true
+  snap remove --purge test-snapd-sh || true
+
+debug: |
+  snap changes
+  cat /home/test/notif.log || true
+
+execute: |
+  changeAfterID() {
+    local OLD_CHANGE="$1"
+    local NEW_CHANGE
+
+    for _ in $(seq 5); do
+      NEW_CHANGE=$(snap changes | tail -n 2 | head -n 1 | awk '{print $1}')
+      if [ "$NEW_CHANGE" -gt "$OLD_CHANGE" ]; then
+        break
+      fi
+
+      snap debug ensure-state-soon
+      sleep 1s
+    done
+
+    if [ "$NEW_CHANGE" -le "$OLD_CHANGE" ]; then
+      echo "expected a change with an id greater than $OLD_CHANGE"
+      exit 1
+    fi
+  }
+
+  test-snapd-sh.sh -c "while [ ! -e stamp ]; do sleep 1; done" &
+  APP_PID="$!"
+  tests.cleanup defer "kill \"$APP_PID\" || true"
+
+  # TODO: test notifications in other distros as well
+  if os.query is-ubuntu && os.query is-classic; then
+    tests.session -u test exec sh -c 'dbus-monitor > notif.log' &
+    tests.cleanup defer "systemctl kill user-12345.slice || true"
+  fi
+
+  # trigger an auto-refresh
+  OLD_CHANGE=$(snap changes | tail -n 2 | head -n 1 | awk '{print $1}')
+  systemctl stop snapd.{service,socket}
+  "$TESTSTOOLS"/snapd-state change-snap-channel test-snapd-sh edge
+  "$TESTSTOOLS"/snapd-state force-autorefresh
+  systemctl start snapd.{socket,service}
+
+  # check that a change was triggered and it's a pre-download
+  changeAfterID "$OLD_CHANGE"
+  if ! retry -n 15 sh -c 'snap changes | tail -n 2 | head -n 1 | grep "Done.*Pre-download tasks for auto-refresh"'; then
```

Comment:
nitpick: Do not think `head -n 1 |` is required. 
Same [here](https://github.com/snapcore/snapd/pull/12609/files#diff-a46c6b8bb9da09f9fcfc4b93a3391d02a6fe263122267ef6dee1611d21bdee6eR93) and [here](https://github.com/snapcore/snapd/pull/12609/files#diff-a46c6b8bb9da09f9fcfc4b93a3391d02a6fe263122267ef6dee1611d21bdee6eR102)

Reply:
Thanks for this, I removed all of them